File: spec/java/src/io/kaitai/struct/spec/TestOpaqueExternalType.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.OpaqueExternalType;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestOpaqueExternalType extends CommonSpec {

    @Test
    public void testOpaqueExternalType() throws Exception {
        OpaqueExternalType r = OpaqueExternalType.fromFile(SRC_DIR + "term_strz.bin");

        assertIntEquals(r.hw().one(), 102);
    }
}

File: spec/rust/tests/test_opaque_external_type.rs
use std::fs;
extern crate kaitai;
use self::kaitai::*;
use rust::formats::opaque_external_type::*;

#[test]
fn test_opaque_external_type() -> KResult<()> {
    let bytes = fs::read("../../src/term_strz.bin").unwrap();
    let _io = BytesReader::from(bytes);
    let r: OptRc<OpaqueExternalType> = OpaqueExternalType::read_into(&_io, None, None)?;

    assert_eq!(*r.one().s1(), "foo");
    assert_eq!(*r.one().s2(), "bar");
    assert_eq!(*r.one().s3(), "|baz@");
    Ok(())
}
